### PR TITLE
Hw3 task3 filter task: bugs were fixed, tests were written

### PR DESCRIPTION
**Pull request by IuliiaLanchuk**

### CODE REVIEW

File: homework3/task03/project_filter/filter.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    """
+    Helper filter class. Accepts a list of single-argument
+    functions that return True if object in list conforms to some criteria
+    """
+
+    def __init__(self, *functions):
+        self.functions = functions
+
+    def apply(self, data):
+        return [item for item in data if all(i(item) for i in self.functions)]
```
Comment (Cjkjvfnby) [inline homework3/task03/project_filter/filter.py]: Does the name `i` suits this task?

File: homework3/task03/tests/test_filter.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+@pytest.mark.parametrize(
+    ("data", "expected_result"),
+    [
+        ({"name": "polly"}, [sample_data[1]]),
+        ({"name": "Bill", "occupation": "was here"}, [sample_data[0]]),
+        ({"name": "Iuliia"}, []),
+    ],
+)
+def test_make_filter(data: Dict[Any, Any], expected_result: List[Any]):
```
Comment (Cjkjvfnby) [inline homework3/task03/tests/test_filter.py]: Why do you need sample_data in the test with the name `test_make_filter`?

File: homework3/task03/tests/test_filter.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    },
+]
+
+
+@pytest.mark.parametrize(
+    ("data", "expected_result"),
+    [
+        ({"name": "polly"}, [sample_data[1]]),
+        ({"name": "Bill", "occupation": "was here"}, [sample_data[0]]),
+        ({"name": "Iuliia"}, []),
```
Comment (Cjkjvfnby) [inline homework3/task03/tests/test_filter.py]: Let's make it as separate tests. I expect more than 3 cases.

File: homework3/task03/tests/test_filter.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+def test_make_filter(data: Dict[Any, Any], expected_result: List[Any]):
+    assert make_filter(**data).apply(sample_data) == expected_result
+
+
+def test_apply():
+    positive_even = Filter(
+        lambda a: a % 2 == 0, lambda a: a > 0, lambda a: isinstance(a, int)
+    )
+    assert positive_even.apply(range(10)) == [2, 4, 6, 8]
+    assert positive_even.apply(range(10)) != [0, 2, 4, 6]
```
Comment (Cjkjvfnby) [inline homework3/task03/tests/test_filter.py]: You have checked that this expression is equal to another expression. This implies that it is not equal to any other. This line has no sense.

**Review by Cjkjvfnby — CHANGES_REQUESTED** [review]

File: homework3/task03/project_filter/filter.py (hunk 1 above)
Comment (IuliiaLanchuk) [reply homework3/task03/project_filter/filter.py]: Agree, fixed

File: homework3/task03/tests/test_filter.py (hunk 2 above)
Comment (IuliiaLanchuk) [reply homework3/task03/tests/test_filter.py]: Do you imply this test should be renamed? Fixed

File: homework3/task03/tests/test_filter.py [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+
+def test_apply_positive_even():
+    positive_even = Filter(
+        lambda a: a % 2 == 0, lambda a: a > 0, lambda a: isinstance(a, int)
+    )
+    assert positive_even.apply(range(10)) == [2, 4, 6, 8]
+
+
+def test_apply_sum_with_7_is_zero():
+    sum_with_7_is_zero = Filter(lambda a: a != 0, lambda a: a + 7 == 0)
```
Comment (Cjkjvfnby) [inline homework3/task03/tests/test_filter.py]: Is any logical relation between this functions `lambda a: a != 0`  `lambda a: a + 7 == 0`?

File: homework3/task03/tests/test_filter.py [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+    assert sum_with_7_is_zero.apply([-7, 13, 6.8, 9, 0]) == [-7]
+
+
+def test_apply_no_data_to_filter():
```
Comment (Cjkjvfnby) [inline homework3/task03/tests/test_filter.py]: It looks like a two tests

test_empty_filter_returns_all_items
test_filter_of_empty_list_return_empty_list

**Review by Cjkjvfnby — APPROVED** [review]

File: homework3/task03/tests/test_filter.py (hunk 6 above)
Comment (IuliiaLanchuk) [reply homework3/task03/tests/test_filter.py]: Yes, thanks, good names:)

File: homework3/task03/tests/test_filter.py (hunk 5 above)
Comment (IuliiaLanchuk) [reply homework3/task03/tests/test_filter.py]: Yes, you are right, we can choose only the second one.

File: homework3/task03/tests/test_filter.py [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+    assert Filter().apply(range(5)) == [0, 1, 2, 3, 4]
+
+
+def test_one_key_value_pair_repeats_in_different_team_members():
+    team = [
+        {
+            "name": "Iuliia",
+            "last_name": "Lanchuk",
+            "hobby": "swimming",
+            "type": "human",
```
Comment (Cjkjvfnby) [inline homework3/task03/tests/test_filter.py:28]: To test with a single key you can remove some other attributes to make it more readable

**Review by Cjkjvfnby — APPROVED** [review]

**Pull request merged**